File: 7-reverse-integer/reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int flag=0;
        if(x<0)
        flag=1;
        x=abs(x);
        long long rev=0;
        while(x>0){
            rev=rev*10+x%10;
            x=x/10;
        }
        if(rev<=INT_MIN || rev>=INT_MAX)
        return 0;
        if(flag==1){
            return -rev;
        }
        return rev;
    }
};

File: 7-reverse-integer/reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(num) {
    let flag=0;
    if(num<0){
        flag=1;
        num=Math.abs(num);
    }
    let rev=0;
    while(num>0){
        rev=rev*10+(num%10);
        num=Math.floor(num/10);
    }
    if(rev<=-Math.pow(2,31) || rev>=Math.pow(2,31)-1){
        return 0;
    }
    if(flag==1)
    return -rev;
    return rev;
};